### PR TITLE
combined memcaching with multi mac support

### PR DESCRIPTION
**Pull request by lathrop-navisite**

### CODE REVIEW

File: staticDHCPd/staticdhcpdlib/config.py [hunk 1]
```diff
@@ -210,6 +212,19 @@
         handleUnknownMAC = conf.handleUnknownMAC
 else:
     handleUnknownMAC = lambda *args, **kwargs : None
+
+if hasattr(conf, 'filterRetrievedDefinitions'):
+    filterRetrievedDefinitions = conf.filterRetrievedDefinitions
+else:
+    def filterRetrievedDefinitions(definitions, *args, **kwargs):
+        if not definitions:
+            return None
```
Comment (lathrop-navisite) [inline staticDHCPd/staticdhcpdlib/config.py:221]: Change to just return definitions, not None

File: staticDHCPd/staticdhcpdlib/databases/generic.py [hunk 2]
```diff
@@ -234,6 +234,8 @@ def _setupCache(self):
                 _logger.warn("PERSISTENT_CACHE was set, but USE_CACHE was not")
             if config.CACHE_ON_DISK:
                 _logger.warn("CACHE_ON_DISK was set, but USE_CACHE was not")
+            if config.MEMCACHED_CACHE:
+                _logger.warn("CACHE_ON_DISK was set, but USE_CACHE was not")
```
Comment (epluntze) [inline staticDHCPd/staticdhcpdlib/databases/generic.py]: This message should use MEMCACHED_CACHE not CACHE_ON_DISK

File: staticDHCPd/staticdhcpdlib/config.py (hunk 1 above)
Comment (epluntze) [inline staticDHCPd/staticdhcpdlib/config.py:221]: If you make that change you should also change this: https://github.com/Navisite/staticdhcpd/pull/5/files#diff-4f4c9f9e5012de9b7f6da0369cc35257R26

Comment (lathrop-navisite) [inline staticDHCPd/staticdhcpdlib/config.py:221]: I do want None. Leaving both as is

File: staticDHCPd/staticdhcpdlib/databases/generic.py (hunk 2 above)
Comment (lathrop-navisite) [inline staticDHCPd/staticdhcpdlib/databases/generic.py]: fixed

File: libpydhcpserver/libpydhcpserver/dhcp.py [hunk 3]
```diff
@@ -68,11 +68,11 @@ class DHCPServer(object):
     """
     _server_address = None #: The IP associated with this server.
     _network_link = None #: The I/O-handler; you don't want to touch this.
-    
-    def __init__(self, server_address, server_port, client_port, pxe_port=None, response_interface=None, response_interface_qtags=None):
+
+    def __init__(self, server_address, server_port, client_port, pxe_port=None, response_interface=None, response_interface_qtags=None, listen_interface=None):
```
Comment (zwerheim) [inline libpydhcpserver/libpydhcpserver/dhcp.py:77]: This new parameter is missing a doc string. See line 86.

File: staticDHCPd/staticdhcpdlib/dhcp.py [hunk 4]
```diff
@@ -400,11 +410,11 @@ class _DHCPServer(libpydhcpserver.dhcp.DHCPServer):
     _database = None #: The database to use for retrieving lease definitions.
     _dhcp_actions = None #: The MACs and the number of actions each has performed, decremented by one each tick.
     _ignored_addresses = None #: A list of all MACs currently ignored, plus the time remaining until requests will be honoured again.
-    
-    def __init__(self, server_address, server_port, client_port, pxe_port, response_interface, response_interface_qtags, database):
+
+    def __init__(self, server_address, server_port, client_port, pxe_port, response_interface, response_interface_qtags, database, listen_interface=None):
```
Comment (zwerheim) [inline staticDHCPd/staticdhcpdlib/dhcp.py:415]: Needs new docstring.

**Comment by lathrop-navisite** [comment]
@flan I implemented as you suggested.
I agree that there should be 1 config option is possible. 
Just a note, on my box socket.SO_BINDTODEVICE is not defined. The "IN" package does have that attribute. That modules is generated from "/usr/include/netinet/in.h" for me. I'm not sure the usage of the IN module across other platforms.

**Pull request closed without merging**